### PR TITLE
Fixes #25825: Deleted OIDC user are reactivated after new provisioning

### PR DESCRIPTION
**Pull request by clarktsiory**
https://issues.rudder.io/issues/25825

* Don't add deleted users back : the `computeUpdatedUserList` method which is reused when "setting existing users" has logic to handle "deleted" and "disabled" user differently, deleted users should not be passed back to existing users (the logic of the method will make already disabled ones remain disabled)
* Add unit tests, 2 tests cases are added : 
  * a simple one to disable Bob
  * another one to check that adding a new user Xavier does not change the state of others : Bob -> disabled, William -> deleted
  Every other changes in the test file is just about moving the values used within the test cases right above the block of the test, the diff cannot just guess it

### CODE REVIEW

**Comment by clarktsiory** [comment]
PR updated with a new commit

**Comment by clarktsiory** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/users/UserRepositoryTest.scala [hunk 1]
```diff
@@ -373,12 +447,62 @@ trait UserRepositoryTest extends Specification with Loggable {
...
+        case u => u
+      }
+    }
+
+    "some user is disabled" >> {
+      val disabled = repo.disable(List("bob"), None, List.empty, traceBobDisabled).runNow
+      (disabled must containTheSameElementsAs(List("bob"))) and (
+        repo.getAll().runNow.toUTC must containTheSameElementsAs(userInfosBobDisabled)
+      )
+    }
```
Comment (clarktsiory) [inline webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/users/UserRepositoryTest.scala:475]: Additional test case no. 1

File: webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/users/UserRepositoryTest.scala [hunk 2]
```diff
@@ -373,12 +447,62 @@ trait UserRepositoryTest extends Specification with Loggable {
...
+        StatusHistory(UserStatus.Active, traceXavierOidc) :: Nil,
+        Json.Obj()
+      )
+    }
+
+    "Some user is added in remote with some other remote users disabled/deleted" >> {
+      // Bob: disabled, William: deleted, adding a user should not change the state of existing ones
+      val isAdded = repo.addUser(AUTH_PLUGIN_NAME_REMOTE, "xavier", traceXavierOidc).runNow
+      isAdded must beTrue and (repo.getAll().runNow.toUTC must containTheSameElementsAs(userInfosXavierOidc))
+    }
```
Comment (clarktsiory) [inline webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/users/UserRepositoryTest.scala:498]: Additional test case no. 2

**Review by VinceMacBuche — APPROVED** [review]

**Comment by VinceMacBuche** [comment]
OK, squash merging this PR

**Pull request merged**